pkgs/by-name/nm/nmgui/package.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  wrapGAppsHook4,
+  gobject-introspection,
+  gtk4,
+  glib,
+  makeDesktopItem,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "nmgui";
+  version = "1.0.0";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "s-adi-dev";
+    repo = "nmgui";
+    tag = "v${version}";
+    hash = "sha256-HS/n40Ng8S5N14DtEH/upwlxdzwCoOEJA40EMdCcLw4=io";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    gtk4
+    glib
+  ];
+
+  desktopItem = makeDesktopItem {
+    name = "nmgui";
+    exec = "nmgui";
+    icon = "network-wireless-symbolic";
+    comment = "GTK4-based Network Manager GUI using nmcli";
+    desktopName = "NM GUI";
+    categories = [
+      "Network"
+      "GTK"
+    ];
+    startupNotify = true;
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/applications,opt/nmgui}
+    # Copy the app files
+    cp -r app $out/opt/nmgui/
+    cp -r ${desktopItem}/share/applications/* $out/share/applications/
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${python3Packages.python}/bin/python3 $out/bin/nmgui \
+      --add-flags "$out/opt/nmgui/app/main.py" \
+      "''${gappsWrapperArgs[@]}"
+  '';
```

```suggestion
  dontWrapGApps = true;

  makeWrapperArgs = [ "\${gappsWrapperArgs[@]}" ];
```
`buildPythonApplication` will already do the wrapping for you, you just have to tell it to also do the gtk wrapping too

---

I've checked the outputs carefully, it is just putting nmgui in /nix/store/q6ipjdm08fddb3gyb2kg1lklh4wp8ars-nmgui-1.0.0/bin - I think we've successfully avoided falling into that trap :)